deferred.go

```diff
@@ -90,6 +99,9 @@ func (d *Deferred) UnmarshalCBOR(br io.Reader) (err error) {
 			return fmt.Errorf("unhandled deferred cbor type: %d", maj)
 		}
 	}
-	d.Raw = buf.Bytes()
+	// Reuse existing buffer. Also, copy to "give back" the allocation in the byte buffer (which
+	// is likely significant).
+	d.Raw = d.Raw[:0]
+	d.Raw = append(d.Raw, buf.Bytes()...)
```

Comment:
Checking for understanding:

In the old code the assignment of `buf.Bytes()` to `d.Raw` brought along the overallocated capacity.  Now by assigning `d.Raw` to the `append` call this is no longer happening.  `d.Raw` keeps its old capacity and the append call will only increase its capacity if the length of `buf.Bytes()` exceeds capacity.

Is this correct?

Reply:
Ah, no. I'm _keeping_ any overallocation from `Raw` (reusing the buffer) but removing any overallocation from the bytes Buffer by copying.

Note: in most  cases, `d.Raw` will be nil. It _won't_ be nil in some special cases where, e.g., we're re-using the same one in a loop. In that case, we don't care about a bit of overallocation as we're _re-using_ space.

The issue with `bytes.Buffer` is that we're over-allocating entirely new space.